### PR TITLE
Add @Secured and JSR250 annotations to 'security-method' sample

### PR DESCRIPTION
**Pull request by marcusdacoregio**

### CODE REVIEW

File: security-method/src/aotTest/java/com/example/security/method/SecurityMethodApplicationAotTests.java [hunk 1]
```diff
@@ -19,22 +19,45 @@ void anonymousCanCallOnlyAnonymousMethod(AssertableOutput output) {
 			assertThat(output)
 					.hasSingleLineContaining("testAnonymous(): protectedService.anonymous() worked as anonymous")
 					.hasSingleLineContaining("testUser(): protectedService.user() failed as anonymous")
-					.hasSingleLineContaining("testAdmin(): protectedService.admin() failed as anonymous");
+					.hasSingleLineContaining("testAdmin(): protectedService.admin() failed as anonymous")
```
Comment (mhalbritter) [inline security-method/src/aotTest/java/com/example/security/method/SecurityMethodApplicationAotTests.java:22]: I would prefer it if `@Secured` and JSR250 are separate tests (if these features are separate code paths in Spring Security). This way we can look at the failing tests to see which features are broken.

File: security-method/src/main/java/com/example/security/method/ProtectedService.java [hunk 2]
```diff
@@ -1,5 +1,10 @@
 package com.example.security.method;
 
+import jakarta.annotation.security.DenyAll;
```
Comment (mhalbritter) [inline security-method/src/main/java/com/example/security/method/ProtectedService.java:3]: These imports are not necessary, are they?

**Comment by mhalbritter** [comment]
Hey @marcusdacoregio , thanks for the PR. I've left some comments.

**Comment by marcusdacoregio** [comment]
Thanks, @mhalbritter.

I've split the tests into multiple tests, one for each way to secure the methods. You may notice that there is a [`catch (IllegalArgumentException)`](https://github.com/spring-projects/spring-aot-smoke-tests/pull/97/files#diff-6e15fbd531f75c74611e979eaf517be68c533e1df5bfbee10e85a92b8651dfe7R102) instead of the usual `AuthenticationException`. I'll check with @jzheaux if that is desired or if we can do something to about that.

**Comment by mhalbritter** [comment]
Thanks for the changes! I'll merge the PR, we can refine if the `IllegalArgumentException` changes to something else.

Regarding the SNAPSHOT version: We have this "problem" in other tests too (e.g. all spring-data tests). When Boot updates to SNAPSHOT, those tests will get green.

**Pull request merged**